test/common/formatter/substitution_formatter_test.cc

```diff
@@ -2336,6 +2337,55 @@ TEST(SubstitutionFormatterTest, ParserSuccesses) {
   }
 }
 
+class EnvFormatterTest : public testing::Test {
+public:
+  ~EnvFormatterTest() override { TestEnvironment::unsetEnvVar("TEST_KEY"); }
+};
+
+TEST_F(EnvFormatterTest, ExistingEnvVars) {
+  TestEnvironment::setEnvVar("TEST_KEY", "test", 1);
+  StreamInfo::MockStreamInfo stream_info;
```

Comment:
Since you are using fixture class here, you can move `stream_info`, `*_headers` and `body `to fixture class and avoid declaring them in test routines.